src/gordon_gcp/plugins/janitor/gpubsub_publisher.py

```diff
@@ -223,13 +224,13 @@ def publish(self, message):
         message['timestamp'] = datetime.datetime.utcnow().isoformat()
         bytes_message = bytes(json.dumps(message), encoding='utf-8')
         future = self.publisher.publish(self.topic, bytes_message)
-        future.add_done_callback(
-            functools.partial(self._message_publish_callback, message))
 
         # collect to make sure everything's cleaned up once done
         self._messages.add(future)
         # TODO (lynn): add metrics.incr/emit call here once aioshumway
         #              is released
+        future.add_done_callback(
+            functools.partial(self._message_publish_callback, message))
```

Comment:
This had to be moved down, in case it completes quickly, it would try to remove itself from the set before it was added.

Reply:
Ah: 'If the future has already completed or been cancelled, fn will be called immediately.'

Reply:
Yep.